### DISCUSSION TITLE
Allow to quickly filter user status in the Users view

### DISCUSSION THREAD

**Discussion by nblock** [keycloak/keycloak#19930] [Ideas]
The Users view in the Admin UI lists all users with their current status as column ("–", "Disabled", …). Some organizations do not delete obsolete users but disable them for compliance reasons. Over time, the Users view is cluttered with many "Disabled" users that are most likely not of interest to the administrator.

A possible solution is to add a toggle to the Users view to quickly filter "Disabled" users. It could be:

* enabled: only active users are displayed (a breaking change to the UI)
* disabled: all users are shown but users may be filtered with a single click (a non-breaking change to the UI)

![image](https://user-images.githubusercontent.com/131119/234203497-7b72afca-0ca2-4682-b9ac-24352b877eef.png)

**Comment by jonkoops** [comment]
ping @xianli123 @ssilvert

**Comment by ssilvert** [comment]
@nblock I agree that we should support this use case.  This should probably be rolled into the new user attribute search:
https://github.com/keycloak/keycloak/issues/17728#issuecomment-1500207677

 @xianli123 What do you think?

**Comment by xianli123** [comment]
@nblock @ssilvert @jonkoops Thanks. That's a good point. We can add 2 options named "Show all users" and "Just show enabled users" as shown in the screenshot below. If users want to search for users based on some filters, there will be some chips to indicate the filters as shown in the second screenshot below. What do you think? Is it easy to implement? 

**1st screen**
 <img width="924" alt="Screenshot 2023-04-26 at 14 52 54" src="https://user-images.githubusercontent.com/53931519/234493681-4b6f7896-1dfb-4c7b-a567-59c56df71019.png">

**2nd screen**
<img width="939" alt="Screenshot 2023-04-26 at 15 01 50" src="https://user-images.githubusercontent.com/53931519/234495563-7a2e4565-5fa0-4cd2-995b-3d34c5d4e40f.png">

**Reply by jonkoops** [reply]
@xianli123 instead of adding new UI elements that take up screen space in the main view, perhaps this could be part of the attribute search (which I think has not yet been implemented)?

**Comment by xianli123** [comment]
@jonkoops Do you mean to consider the `Status: Enabled/Disabled` as an attribute of users as in the screenshot below?

<img width="855" alt="Screenshot 2023-04-26 at 15 57 24" src="https://user-images.githubusercontent.com/53931519/234509703-e08ec033-461b-41f2-9b9a-0faf5608d0cb.png">

**Reply by jonkoops** [reply]
Yes, pretty much exactly like that :+1:

**Reply by xianli123** [reply]
Ok, let's wait for others' thoughts. 🤓

**Reply by andreas-blaettlinger** [reply]
I like the idea itself but is the user attribute search the right name for this? I mean the user status is not a attribute, right? We could rename it to something like `Detailed search` (instead of `Attribute search`) and a general `Search by` (instead of `Search by attributes`).

**Reply by jonkoops** [reply]
Yeah, I think we can work on labeling this differently that's not a problem. @xianli123 do you happen to know if we already have the attribute search implemented? I cannot find it in the UI.

**Reply by xianli123** [reply]
Yes, I knew some implements have been done or working on. FYI - https://github.com/keycloak/keycloak/pull/19574. The contributor also is [andreas-blaettlinger](https://github.com/andreas-blaettlinger).

**Reply by jonkoops** [reply]
Ah, yes I still need to get on reviewing that work. I'll get into that.

**Reply by xianli123** [reply]
> I like the idea itself but is the user attribute search the right name for this? I mean the user status is not a attribute, right? We could rename it to something like `Detailed search` (instead of `Attribute search`) and a general `Search by` (instead of `Search by attributes`).

With regard to the wording, we can ask for the doc writer's help. Thanks for your suggestions! 🙏

**Comment by sweh** [comment]
This feature is essential for auditing purposes. You need to be able to see at a glance which users can currently log in and which can't. Is a status filter planned or already implemented?
